SingularityUI/app/components/taskSearch/TaskSearch.jsx

```diff
@@ -1,166 +1,181 @@
 import React from 'react';
+import { connect } from 'react-redux';
+import classNames from 'classnames';
+import { FetchTaskSearchParams } from '../../actions/api/history';
+
+import Breadcrumbs from '../common/Breadcrumbs';
+import TasksTable from './TasksTable';
+import TaskSearchFilters from './TaskSearchFilters';
+import Glyphicon from '../common/atomicDisplayItems/Glyphicon';
+import JSONButton from '../common/JSONButton';
 import Utils from '../../utils';
-import Enums from './Enums';
-import TaskSearchForm from './TaskSearchForm';
-import DisplayResults from './DisplayResults';
-import Header from './Header';
-
-let TaskSearch = React.createClass({
-
-    countChoices: [5, 10, 25],
-
-    defaultCount: 10,
-
-    defaultSortDirection: 'DESC',
-
-    getInitialState() {
-        return {
-            form: {
-                requestId: this.props.initialRequestId || '',
-                deployId: this.props.initialDeployId || '',
-                host: this.props.initialHost || '',
-                lastTaskStatus: this.props.initialTaskStatus || '',
-                startedBefore: this.props.initialStartedBefore || '',
-                startedAfter: this.props.initialStartedAfter || ''
-            },
-            sortDirection: this.props.initialSortDirection || this.defaultSortDirection,
-            queryParams: {
-                requestId: this.props.initialRequestId || '',
-                deployId: this.props.initialDeployId || '',
-                host: this.props.initialHost || '',
-                lastTaskStatus: this.props.initialTaskStatus || '',
-                startedBefore: this.props.initialStartedBefore || '',
-                startedAfter: this.props.initialStartedAfter || ''
-            },
-            pageNumber: 1,
-            count: this.props.initialCount || this.defaultCount,
-            hasDoneAnySearch: false
-        };
-    },
-
-    handleSubmit(event) {
-        event.preventDefault();
-        return this.setState({
-            queryParams: this.state.form,
-            pageNumber: 1, // If you narrow down your search you most likely want to go back to page 1
-            hasDoneAnySearch: true
+
+class TaskSearch extends React.Component {
+
+  static propTypes = {
+    requestId: React.PropTypes.string,
+    fetchTaskHistory: React.PropTypes.func.isRequired,
+    request: React.PropTypes.object,
+    taskHistory: React.PropTypes.array
+  }
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      filter: {
+        requestId: props.requestId,
+        page: 1,
+        count: TaskSearch.TASKS_PER_PAGE
+      },
+      disableNext: false,
+      loading: false
+    };
```

Comment:
Long term, task search results should be true permalinks (i.e. have all this state data in the URL), and that would make for a good argument to move this stuff in to Redux (I'm using https://github.com/reactjs/redux/issues/1287#issuecomment-175351978 as a guide). But since permalinks don't exist in the original implementation, I'm okay with circling back on this.